harp/reader.py

```diff
@@ -44,13 +45,37 @@ def __init__(
         self.read = read
 
 
+class RegisterMap(UserDict[str, RegisterReader]):
+    _address_map: dict[int, str]
```

Comment:
I think this should be a map from `int` to `RegisterReader`:

```suggestion
    _address_map: dict[int, RegisterReader]
```